src/fpm_compiler.F90

```diff
@@ -953,16 +953,31 @@ subroutine make_archive(self, output, args, log_file, stat)
     integer, intent(out) :: stat
 
     if (self%use_response_file) then
-        !$omp critical
         call write_response_file(output//".resp" , args)
         call run(self%ar // output // " @" // output//".resp", echo=self%echo, &
             &  verbose=self%verbose, redirect=log_file, exitstat=stat)
+#if defined(_WIN32) && defined(__GFORTRAN__)
+        call delete_file_win32(output//".resp")
+#else
         call delete_file(output//".resp")
-        !$omp end critical
+#endif
     else
         call run(self%ar // output // " " // string_cat(args, " "), &
             & echo=self%echo, verbose=self%verbose, redirect=log_file, exitstat=stat)
     end if
+#if defined(_WIN32) && defined(__GFORTRAN__)
```

Comment:
I think this is prone to fail as the `_WIN32` macro is not declared in GFortran. cc Minh @minhqdao

Reply:
Yes, you can try to define or not define `_WIN32` and compile it with gfortran to see if it works. It didn't for me on MacOS.

Reply:
Yes, I just tried it, it'seems that GFortran on Windows does not have a `_WIN32` macro defined. 
How about the following code, using `get_os_type`?
```diff
diff --git a/src/fpm_compiler.F90 b/src/fpm_compiler.F90
index fb6d3d65..a38597dd 100644
--- a/src/fpm_compiler.F90
+++ b/src/fpm_compiler.F90
@@ -956,16 +956,17 @@ subroutine make_archive(self, output, args, log_file, stat)     
         call write_response_file(output//".resp" , args)
         call run(self%ar // output // " @" // output//".resp", echo=self%echo, &     
             &  verbose=self%verbose, redirect=log_file, exitstat=stat)
-#if defined(_WIN32) && defined(__GFORTRAN__)
-        call delete_file_win32(output//".resp")
-#else
-        call delete_file(output//".resp")
-#endif
+        if (get_os_type() == OS_WINDOWS) then
+            call delete_file_win32(output//".resp")
+        else
+            call delete_file(output//".resp")
+        end if
+
     else
         call run(self%ar // output // " " // string_cat(args, " "), &
             & echo=self%echo, verbose=self%verbose, redirect=log_file, exitstat=stat)
     end if
-#if defined(_WIN32) && defined(__GFORTRAN__)
+
     contains
         subroutine delete_file_win32(file)
             character(len=*), intent(in) :: file
@@ -977,7 +978,6 @@ subroutine make_archive(self, output, args, log_file, stat)       
                 close(unit, status='delete', iostat=iostat)
             end if
         end subroutine delete_file_win32
-#endif
 end subroutine make_archive
```

Reply:
That seems like the way to go for now.aybe we should have fpm define macros for this.

Reply:
I would say this is perfectly fine if it compiles

Reply:
In fact, only Windows-fpm will use `* .resp` files (`self%use_response_file` is true on Windows). 
So instead of using `get_os_type` to judge OS type, we can directly use `delete_file_win32` here.